### ISSUE TITLE
Cython 3 breaks cythonization of _reweight.pyx

### ISSUE THREAD

**Issue by jeremyleung521** [westpa/westpa#323]
### Prerequisites

* [x] Can you reproduce the problem in safe mode?
* [x] Are you running the latest version?
* [x] Are you reporting to the correct repository?
* [x] Did you perform a cursory search?

For more information, see the `CONTRIBUTING` guide.

**Describe the bug.** A clear and concise description of what the bug is.  
Cython 3 (released July 17, 2023) broke cythonization of `_reweight.pyx`. I can prevent this on our end, but a [similar problem](https://github.com/mdtraj/mdtraj/issues/1800) is also happening to our upstream dependency (mdtraj). Already reported that... will have a PR for this up soon.

**To Reproduce.** Steps to reproduce the behavior:  
1. `python -m pip install westpa --no-binary :all:` in an environment without westpa.

**Expected behavior.** A clear and concise description of what you expected to happen.  
Successful installation.

**Actual behavior.** A clear and concise description of what actually happens.  
```
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'error'
    error: subprocess-exited-with-error
    
    × Getting requirements to build wheel did not run successfully.
    │ exit code: 1
    ╰─> [66 lines of output]
        
        Error compiling Cython file:
        ------------------------------------------------------------
        ...
        from .linalg cimport cython_blas, cython_lapack
        ^
        ------------------------------------------------------------
        
        /tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/scipy/linalg.pxd:1:0: 'scipy/linalg/linalg.pxd' not found
        
        Error compiling Cython file:
        ------------------------------------------------------------
        ...
        from .linalg cimport cython_blas, cython_lapack
        ^
        ------------------------------------------------------------
        
        /tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/scipy/linalg.pxd:1:0: 'scipy/linalg/linalg/cython_blas.pxd' not found
        
        Error compiling Cython file:
        ------------------------------------------------------------
        ...
        from .linalg cimport cython_blas, cython_lapack
        ^
        ------------------------------------------------------------
        
        /tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/scipy/linalg.pxd:1:0: 'scipy/linalg/linalg/cython_lapack.pxd' not found
        warning: src/westpa/core/reweight/_reweight.pyx:351:13: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
        warning: src/westpa/core/reweight/_reweight.pyx:351:21: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
        warning: src/westpa/core/reweight/_reweight.pyx:351:32: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
        warning: src/westpa/core/reweight/_reweight.pyx:352:15: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
        warning: src/westpa/core/reweight/_reweight.pyx:352:24: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
        warning: src/westpa/core/reweight/_reweight.pyx:352:35: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
        warning: src/westpa/core/reweight/_reweight.pyx:352:46: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
        warning: src/westpa/core/reweight/_reweight.pyx:352:54: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
        Compiling src/westpa/fasthist/_fasthist.pyx because it changed.
        Compiling src/westpa/trajtree/_trajtree.pyx because it changed.
        Compiling src/westpa/mclib/_mclib.pyx because it changed.
        Compiling src/westpa/core/binning/_assign.pyx because it changed.
        Compiling src/westpa/core/kinetics/_kinetics.pyx because it changed.
        Compiling src/westpa/core/reweight/_reweight.pyx because it changed.
        [1/6] Cythonizing src/westpa/core/binning/_assign.pyx
        [2/6] Cythonizing src/westpa/core/kinetics/_kinetics.pyx
        [3/6] Cythonizing src/westpa/core/reweight/_reweight.pyx
        Traceback (most recent call last):
          File "/opt/python/cp39-cp39/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
            main()
          File "/opt/python/cp39-cp39/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
            json_out['return_val'] = hook(**hook_input['kwargs'])
          File "/opt/python/cp39-cp39/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
            return hook(config_settings)
          File "/tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
            return self._get_build_requires(config_settings, requirements=['wheel'])
          File "/tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
            self.run_setup()
          File "/tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 487, in run_setup
            super(_BuildMetaLegacyBackend,
          File "/tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 338, in run_setup
            exec(code, locals())
          File "<string>", line 155, in <module>
          File "<string>", line 59, in extensions
          File "/tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/Cython/Build/Dependencies.py", line 1134, in cythonize
            cythonize_one(*args)
          File "/tmp/pip-build-env-awus4odb/overlay/lib/python3.9/site-packages/Cython/Build/Dependencies.py", line 1301, in cythonize_one
            raise CompileError(None, pyx_file)
        Cython.Compiler.Errors.CompileError: src/westpa/core/reweight/_reweight.pyx
```

**WESTPA version and platform.** Which version you're running, on what OS, and how it was built (e.g. conda, system libraries, manually installed dependencies)  

**Additional context.** Add any other context about the problem here. If necessary config files are required to reproduce the bug, attach the files here here.

**Comment by jeremyleung521** [comment]
Was finally able to isolate and reproduce this error. It's an import order problem only on Linux. Easy to fix on our end, but up to scipy to fix it for good. I posted this to scipy. #324 fixes this.
https://github.com/scipy/scipy/issues/18909

**Comment by jeremyleung521** [comment]
Fixed, Merged and Released as part of v2022.05!
